cmd/kubeadm/app/phases/addons/proxy/manifests.go

```diff
@@ -80,6 +80,7 @@ spec:
         command:
         - /usr/local/bin/kube-proxy
         - --config=/var/lib/kube-proxy/{{ .ProxyConfigMapKey }}
+        - --hostname-override=$NODE_NAME
```

Comment:
I think the syntax needs to be `$(NODE_NAME)` (with the parens) to be substituted by kubelet.  Otherwise it's asking a shell somewhere down the line to do the substitution - and I don't think there's a shell involved in this specific example.  Indeed, if it _does_ also work as you've written, then I'm kind of alarmed, since I think that means the image entrypoint(?) is re-evaluating its arguments and hence introducing a bunch of shell escaping / word-splitting bugs.

My test, just to confirm I wasn't crazy (on my local k8s 1.9 cluster):
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  containers:
    - name: test
      image: busybox
      command:
        - echo
        - A=$FOO
        - B=$(FOO)
        - C=${FOO}
      env:
        - name: FOO
          value: foo
```

=>

```
A=$FOO B=foo C=${FOO}
```

Reply:
mine was substituted. aka A==FOO . @neolit123 and I did talk about what the proper syntax was prior to this. But I can update. running a build and will build a cluster with the `(...)`

Reply:
thanks @anguslees 
i suggested the variable without `( )`, but wasn't 100% sure about it.

@Klaven mentioned that this works in his tests, but we should probably use `( ... )`.

>  Indeed, if it does also work as you've written, then I'm kind of alarmed, since I think that means the image entrypoint(?) is re-evaluating its arguments and hence introducing a bunch of shell escaping / word-splitting bugs.

yes this is puzzling. i'd suggest what we just get it right under `command`, as it's unlikely we will have time to investigate further. too many items to look for 1.13.

Reply:
ok, @anguslees  before I destroyed the old cluster I checked again. `--hostname-override` != `$NODE_NAME` as this would imply. building new cluster now. I'm good with putting the `(...)` in if it works the same. will update you when i'm finished testing.

Reply:
by the way, do you have a good way of automated testing of this? I wanted too.... but ran out of time.

Reply:
IMO, integration or e2e for this are out of scope for this PR.

Reply:
yep, e2e is out of scope imo.

It's clear that there is nowhere (or not much) in the current e2e tests that exercises hostname != node name, since there are frequent bugs around this.  It would be great to change the hostnames/node-names for _all_ the e2e tests, but that's quite separate to this PR.